src/commonMain/kotlin/org/kosat/Incremental.kt

```diff
@@ -0,0 +1,87 @@
+package org.kosat
+
+import kotlin.math.abs
+
+class Incremental(initClauses: MutableList<MutableList<Int>>, initNumber: Int = 0) : CDCL(initClauses, initNumber) {
```

Comment:
Do not abbreviate without a reason. "init" is commonly a verb, but here you need a noun `initialClauses`.
See the discussion above about the inheritance and about the secondary ctor/factory method.